### PR TITLE
Show error message in audio overlay

### LINKED ISSUE #4383: Recording overlay shows errors
**Acceptance**

- [x] Shows recording start errors just for background (pop up is also there)

- [x] Shows warning when recording disabled

### PR DESCRIPTION
**Pull request by seadowg**
Closes #4383
~~Blocked by #4399~~

#### What has been done to verify that this works as intended?

Added new tests and verified manually.

#### Why is this the best possible solution? Were any other approaches considered?

Comments inline.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

This makes a few changes to audio recording errors in general so good to check those (although there are tests covering this).

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/audio/AudioRecordingControllerFragment.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 import static android.view.View.GONE;
 import static android.view.View.VISIBLE;
 import static org.odk.collect.android.utilities.DialogUtils.showIfNotShowing;
-import static org.odk.collect.android.utilities.LiveDataUtils.zip3;
+import static org.odk.collect.android.utilities.LiveDataUtils.zip4;
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/audio/AudioRecordingControllerFragment.java]: 😱😱😱😱😱😱 It keeps growing!

File: shared/src/main/java/org/odk/collect/shared/livedata/LiveDataUtils.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+import kotlin.Triple;
+
+public class LiveDataUtils {
+
+    private LiveDataUtils() {
+
+    }
+
+    public static <T, U, V> LiveData<Triple<T, U, V>> zip3(LiveData<T> one, LiveData<U> two, LiveData<V> three) {
```
Comment (lognaturel) [inline shared/src/main/java/org/odk/collect/shared/livedata/LiveDataUtils.java:14]: None of the `zip3` components are in use, right? I'd remove them and re-add if needed.

**Review by lognaturel — APPROVED** [review]

File: shared/src/main/java/org/odk/collect/shared/livedata/LiveDataUtils.java (hunk 2 above)
Comment (seadowg) [reply shared/src/main/java/org/odk/collect/shared/livedata/LiveDataUtils.java:14]: Will do

**Comment by kkrawczyk123** [comment]
@seadowg I have done a regression check on background recording feature on all my devices. I haven't noticed any potential regressions and I also can not see changes according to error messages. Is there anything more specific that I should check?

**Comment by seadowg** [comment]
@kkrawczyk123 no that sounds good!

**Comment by kkrawczyk123** [comment]
Verified on Androids 5.1, 7.0, 8.1, 9.0 and 10.

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Pull request merged**